src/detect-ipopts.c

```diff
@@ -32,10 +32,6 @@
 #include "detect-ipopts.h"
 #include "util-unittest.h"
 
-#define PARSE_REGEX "\\S[A-z]"
-
-static DetectParseRegex parse_regex;
```

Comment:
For me, removal of PCRE is rather part of the misc cleanup than the fix, which you get right by using `BIT_U16` 

cf https://github.com/OISF/suricata/pull/10700#discussion_r1543658839